lib/internal/modules/cjs/loader.js

```diff
@@ -85,6 +85,17 @@ const relativeResolveCache = Object.create(null);
 
 let requireDepth = 0;
 let statCache = null;
+
+function enrichCJSError(err) {
+  if (
+    err.message.startsWith('SyntaxError: Unexpected token export') ||
+    (/(?:import).*\n{1}\s*\^{9}/mi).test(err.stack)) {
+    err.stack = '\nNote: To load an ES module using --experimental-modules ' +
```

Comment:
Leading `\n` probably not needed here? Or is it?

Reply:
This is to separate it from the experimental modules notice and not look line one body

```
(node:27712) ExperimentalWarning: The ESM module loader is experimental.

Note: To load an ES module set "type": "module" in the package.json or use the .mjs extension.
```

Reply:
The warnings are emitted separately to the errors. The same logic would apply to every warning and every error then. So the newline should be removed.

Reply:
will work by defining with `let` and fn outside the try-catch it the variable is visible to where its used.

Reply:
reworked this a little bit

Reply:
great, thanks.